colcon_defaults/argument_parser/defaults.py

```diff
@@ -24,6 +25,22 @@
     'line arguments (default: $COLCON_HOME/defaults.yaml)')
 
 
+def deep_update(source, overrides):
+    """
+    Update a nested dictionary or similar mapping.
+
+    Modifies ``source`` in place with values from ``overrides``. Taken from
+    https://stackoverflow.com/a/30655448
+    """
+    for key, value in overrides.items():
+        if isinstance(value, collections.abc.Mapping) and value:
+            returned = deep_update(source.get(key, {}), value)
+            source[key] = returned
+        else:
+            source[key] = overrides[key]
+    return source
```

Comment:
Let's drop the return value since the docs state that `source` is modified in place.
```suggestion
```